File: packages/php/src/Resources/Pricing.php
<?php

declare(strict_types=1);

namespace Opensms\Resources;

use Opensms\Models;
use Opensms\Page;
use Opensms\Transport;

/**
 * The `pricing` resource: your price list.
 * Accessed as `$client->pricing`.
 */
final class Pricing
{
    /** @internal */
    public function __construct(private readonly Transport $transport)
    {
    }

    /**
     * @param array{product?: string, country?: string} $params product: sms|lookup|number_monthly
     * @return array<string, mixed>
     */
    public function get(array $params = []): array
    {
        return Models::object($this->transport->request('GET', '/v1/pricing', Models::map($params, Models::PRICING_QUERY, [], true)));
    }
}

File: packages/java/src/main/java/io/opensms/Pricing.java
package io.opensms;

import io.opensms.models.PriceList;

/** Your effective prices. Accessed as {@code client.pricing()}. */
public final class Pricing {

    private final ApiTransport t;

    Pricing(ApiTransport t) {
        this.t = t;
    }

    /** @return the SMS price list for every country. */
    public PriceList get() {
        return get(null, null);
    }

    /**
     * @param product {@code sms} (default), {@code lookup} or {@code number_monthly}, or {@code null}.
     * @param country ISO2 country, or {@code null} for all.
     * @return the price list.
     */
    public PriceList get(String product, String country) {
        return t.get("/v1/pricing", Query.of().add("product", product).add("country", country),
                ApiTransport.type(PriceList.class));
    }
}
